Makefile

```diff
@@ -14,30 +16,40 @@ build/aws-stack.json: $(wildcard templates/*.yml)
 	bundle exec cfoo $^ > $@
 
 setup:
-	which bundle || gem install bundler --no-ri --no-rdoc
-	bundle install --path vendor/bundle
+	bundle check || ((which bundle || gem install bundler --no-ri --no-rdoc) && bundle install --path vendor/bundle)
+	cp config.json.example config.json
 
 clean:
-	-rm build/*
-	-rm templates/mappings.yml
+	-rm -f build/*
+	-rm -f templates/mappings.yml
 
 build-ami:
 	cd packer/; packer build buildkite-ami.json
 
 upload: build/aws-stack.json
-	aws s3 sync --acl public-read build s3://buildkite-aws-stack/
+	aws s3 sync --acl public-read build s3://${BUILDKITE_STACK_BUCKET}/
 
 create-stack: templates/mappings.yml build/aws-stack.json
 	aws cloudformation create-stack \
 	--output text \
-	--stack-name buildkite-$(shell date +%Y-%m-%d-%H-%M) \
+	--stack-name buildkite \
 	--disable-rollback \
 	--template-body "file://${PWD}/build/aws-stack.json" \
 	--capabilities CAPABILITY_IAM \
-	--parameters '$(shell cat config.json)'
+	--parameters '$(shell cat config.json)' \
+	--profile "$(AWS_PROFILE)"
```

Comment:
AFAIK, the aws-cli will read the AWS_PROFILE env anyway, so this shouldn't be needed.